src/udev/udev-builtin-net_id.c

```diff
@@ -161,6 +167,67 @@ static struct udev_device *skip_virtio(struct udev_device *dev) {
         return parent;
 }
 
+static int get_virtfn_info(struct udev_device *dev, struct netnames *names, struct virtfn_info *vf_info) {
+        struct udev *udev;
+        const char *physfn_link_file;
+        _cleanup_free_ char *physfn_pci_syspath = NULL;
+        _cleanup_free_ char *virtfn_pci_syspath = NULL;
+        struct dirent *dent;
+        _cleanup_closedir_ DIR *dir = NULL;
+        struct virtfn_info vf_info_local = {};
+        int rc;
+
+        udev = udev_device_get_udev(names->pcidev);
+        if (!udev)
+                return -ENOENT;
+        /* Check if this is a virtual function. */
+        physfn_link_file = strjoina(udev_device_get_syspath(names->pcidev), "/physfn");
+        rc = chase_symlinks(physfn_link_file, NULL, 0, &physfn_pci_syspath);
+        if (rc <= 0)
+                return (rc < 0) ? rc : -ENOENT;
```

Comment:
This check is a bit strange, since you check the same thing twice kinda, inside of eahc other. please simplify as:

```c
if (rc < 0)
        return rc;
if (rc == 0)
        return -ENOENT;
```

That said, note that unless you specify CHASE_NONEXISTANT as flags this will never return 0, hence you might as well drop the latter two lines...